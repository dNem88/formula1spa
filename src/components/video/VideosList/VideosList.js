import React, {Fragment} from 'react'
import Spinner from '../../common/spinner/Spinner'
import styles from './videosList.module.css'
import Video from '../../common/video/Video'
import VideoInfo from '../../common/videoInfo/VideoInfo'

function VideosList(props) {

    return (
        <Fragment>
             {props.videos? props.videos.map(x => {
                 return <div key={x._id} className={styles['video-wrapper']}>
                            <Video url={x.url}/>
                            <VideoInfo  _id={x._id} title={x.title}/>
                        </div>
            }) : <Spinner/>}
        </Fragment>
       
    )

}

export default VideosList