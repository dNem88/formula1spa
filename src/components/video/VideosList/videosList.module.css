.video-wrapper{
    width: 30vw;
    height: 28vw;
    margin-bottom: 1.5vw;
}

.video-wrapper > :nth-child(1) {
    width: 100%;
    height: 65%;
    border-style: none;
}
.video-wrapper > :nth-child(2) {
    width: 100%;
    height: 35%;
    margin-top: 0;
}

@media only screen and (max-width: 600px) {
    .video-wrapper{
    width: 45vw;
    height: 70vw;
    margin-bottom: 1.5vw;
    }
    .video-wrapper > :nth-child(1) {
    
    height: 50%;
    border-style: none;
     }
.video-wrapper > :nth-child(2) {
    
    height: 50%;
    margin-top: 0;
   
}
}

