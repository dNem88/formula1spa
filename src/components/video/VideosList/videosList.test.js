import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import VideosList from './VideosList'


jest.mock('../../common/video/Video', () => () => 'VideoMock')
jest.mock('../../common/videoinfo/VideoInfo', () => () => 'VideoInfoMock')

describe('Testing VideosList', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <VideosList videos={[{_id: '1'}]}/>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.children.length).toBe(2)
        expect(snapshot.type).toBe('div')
        expect(snapshot.children[0]).toBe('VideoMock')
        expect(snapshot.children[1]).toBe('VideoInfoMock')
        
    })
})