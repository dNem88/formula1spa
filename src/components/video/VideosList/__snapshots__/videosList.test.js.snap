// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideosList Render Properly 1`] = `
<div
  className="video-wrapper"
>
  VideoMock
  VideoInfoMock
</div>
`;
